### DISCUSSION TITLE
query view collection with dynamic parameters

### DISCUSSION THREAD

**Discussion by rezk2ll** [pocketbase/pocketbase#5051] [Q&A] [answered]
I want to achieve the following using view collection

```sql
SELECT
  "cats" as id,
  count(animals.id) as total
FROM animals
WHERE animals.type = :dynamic_parameter
```

where `:dynamic_parameter` is provided when fetching the collection using the JavaScript SDK.

**Comment by ganigeorgiev** [comment, marked as answer]
This was discussed several times before and it is not supported at the moment.

Your description doesn't contain enough details to understand the desired outcome but if you need a count per `animals.type` you can use `GROUP BY` (pseudo-query):

```sql
SELECT
  animals.type as id,
  count(animals.id) as total
FROM animals
GROUP BY animals.type
```

Then on the client-side your can filter by the `type` field:

```js
await pb.collection("yourView").getOne("yourType")
```

(_edit: updated to use the `type` as id_)

**Reply by rezk2ll** [reply]
thanks @ganigeorgiev,
what if the `GROUP BY` solution doesn't apply ( for example a date field )?
the only solution would be creating a custom route?

**Reply by ganigeorgiev** [reply]
> the only solution would be creating a custom route?

Yes, if a regular view doesn't suffice your use case, the only available option at the moment is a custom route.
